### ISSUE TITLE
model.generate() produces different outputs with padding for flan-t5-small

### ISSUE THREAD

**Issue by achartier** [huggingface/transformers#36461]
### System Info

- `transformers` version: 4.49.0
- Platform: Linux-5.15.0-133-generic-x86_64-with-glibc2.39
- Python version: 3.12.3
- Huggingface_hub version: 0.28.1
- Safetensors version: 0.5.2
- Accelerate version: 1.4.0
- Accelerate config:    not found
- DeepSpeed version: not installed
- PyTorch version (GPU?): 2.6.0a0+ecf3bae40a.nv25.01 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>
- Using GPU in script?: no
- GPU type: NVIDIA A30

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
from transformers import AutoModelForSeq2SeqLM, AutoTokenizer

model = AutoModelForSeq2SeqLM.from_pretrained("google/flan-t5-small")
tokenizer = AutoTokenizer.from_pretrained("google/flan-t5-small")

input_texts = [
    "Translate to French: Hello, how are you?",
    "Summarize: The quick brown fox jumps over the lazy dog."
]

inputs = tokenizer(input_texts, return_tensors="pt", padding=True, truncation=True)
outputs = model.generate(**inputs, max_new_tokens=50)

for i, output in enumerate(outputs):
    decoded_output = tokenizer.decode(output, skip_special_tokens=True)
    print(f"Input {i + 1}: {input_texts[i]}")
    print(f"Output {i + 1}: {decoded_output}\n")
```

Output with transformers 4.49.0:
```
Input 1: Translate to French: Hello, how are you?
Output 1:

Input 2: Summarize: The quick brown fox jumps over the lazy dog.
Output 2: The brown fox jumps over the dog.
```

I found references to similar issues (https://github.com/huggingface/transformers/issues/28385), but this one seems to be a regression between 4.48.3 and 4.49.0. I tried combinations of right and left padding with `use_cache`, but no luck.

Interestingly, `t5-small` model does not show this regression.

### Expected behavior

Output with transformers 4.48.3:
```
Input 1: Translate to French: Hello, how are you?
Output 1: Hello, c'est-ce que vous êtes?

Input 2: Summarize: The quick brown fox jumps over the lazy dog.
Output 2: The brown fox jumps over the dog.
```

**Comment by zucchini-nlp** [comment]
Duplicate of https://github.com/huggingface/transformers/issues/36293. It was fixed on main branch, so you can upgrade `transformers`

**Comment by achartier** [comment]
Thanks! I can confirm it is indeed fixed in the main branch.
